pkg/api/v20210131preview/openshiftcluster_validatestatic.go

```diff
@@ -228,6 +227,18 @@ func (sv *openShiftClusterStaticValidator) validateMasterProfile(path string, mp
 	if sr.SubscriptionID != sv.r.SubscriptionID {
 		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path+".subnetId", "The provided master VM subnet '%s' is invalid: must be in same subscription as cluster.", mp.SubnetID)
 	}
+	if mp.DiskEncryptionSetID != "" {
+		if !validate.RxDiskEncryptionSetID.MatchString(mp.DiskEncryptionSetID) {
+			return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path+".diskEncryptionSetId", "The provided master disk encryption set '%s' is invalid.", mp.DiskEncryptionSetID)
+		}
+		desr, err := azure.ParseResourceID(mp.DiskEncryptionSetID)
+		if err != nil {
+			return err
+		}
+		if desr.SubscriptionID != sv.r.SubscriptionID {
+			return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path+".diskEncryptionSetId", "The provided master disk encryption set '%s' is invalid: must be in same subscription as cluster.", mp.DiskEncryptionSetID)
+		}
```

Comment:
I think Keyvault with DiskSet has to be in the same region too. So would be good to check it too.

Reply:
Will do in the dynamic validator, see https://github.com/Azure/ARO-RP/pull/1595#discussion_r668608315

Resolvign this one.